### DISCUSSION TITLE
Mcc

### DISCUSSION THREAD

**Discussion by JouSF-1409** [xumi1993/seispy#58] [General]
最近在读源码，读到mcc功能。有几个小问题：

1. seispy代码中很多数值信号的操作，像是互相关什么的都只用fft转换下计算得到。并不使用scipy或numpy的互相关函数。是有什么特别的考量吗，效率、版本兼容或是什么的。还是单纯的习惯。
2. 最近想基于复现一些互相关提相对到时的论文。mcc现有的功能和Lupei Zhu 的Scc有多少差别。Zhu老师的脚本是perl+C的，看得头晕脑胀。

**Comment by xumi1993** [comment]
1,你说的很对，就是因为频率域计算效率比较高。其次每个波形只做一次fft重复调用也可以提高计算效率。
2, 我不是很清楚你所说的scc是什么方法。

**Reply by JouSF-1409** [reply]
1. 这个重复fft很有道理，学习了。
2. 仔细看了下源码，scc是滑动窗互相关，具体的到时提取是和perl脚本结合的。
3. 想了解下seispy 3DCCP的处理流程，该从哪个脚本开始读？

**Reply by xumi1993** [reply]
可以参考这个脚本[seispy/ccp3d.py](https://github.com/xumi1993/seispy/blob/0cac09e626b43971be016439f20be54505a63b80/seispy/ccp3d.py)
